src/osd/OSD.cc

```diff
@@ -8081,10 +8087,12 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
   }
   map_lock.lock();
 
+  ceph_assert(first <= last);
+
   bool do_shutdown = false;
   bool do_restart = false;
   bool network_error = false;
-  OSDMapRef osdmap;
```

Comment:
with the `ceph_assert(first <= last);` do we still need to initialize the osdmap?

Reply:
initialize the osdmap brings it back to the same logic as before 80da5f9a987c6a48b93f25228fdac85890013520
I think that is preferable.
And the ceph_assert is a sort of documentation

Reply:
yeah that make sense